第十回課題

iijima-pixcel opened the pull request:
new file:   img/lesson10/l10albsecurity.png
	new file:   img/lesson10/l10albtarget.png
	new file:   img/lesson10/l10ap-stack-source.png
	new file:   img/lesson10/l10ec2.png
	new file:   img/lesson10/l10ec2security.png
	new file:   img/lesson10/l10net-stack-source.png
	new file:   img/lesson10/l10rds.png
	new file:   img/lesson10/l10rdssecurity.png
	new file:   img/lesson10/l10s3.png
	new file:   img/lesson10/l10security-stack-source.png
	new file:   img/lesson10/l10vpc.png
	new file:   l10cloud-formation/lesson10ap.yml
	new file:   l10cloud-formation/lesson10network.yml
	new file:   l10cloud-formation/lesson10security.yml
	new file:   lesson10.md

iijima-pixcel committed:
new file:   img/lesson10/l10alb.png
	new file:   img/lesson10/l10albsecurity.png
	new file:   img/lesson10/l10albtarget.png
	new file:   img/lesson10/l10ap-stack-source.png
	new file:   img/lesson10/l10ec2.png
	new file:   img/lesson10/l10ec2security.png
	new file:   img/lesson10/l10net-stack-source.png
	new file:   img/lesson10/l10rds.png
	new file:   img/lesson10/l10rdssecurity.png
	new file:   img/lesson10/l10s3.png
	new file:   img/lesson10/l10security-stack-source.png
	new file:   img/lesson10/l10vpc.png
	new file:   l10cloud-formation/lesson10ap.yml
	new file:   l10cloud-formation/lesson10network.yml
	new file:   l10cloud-formation/lesson10security.yml
	new file:   lesson10.md

iijima-pixcel committed:
modified:   l10cloud-formation/lesson10ap.yml
	modified:   l10cloud-formation/lesson10network.yml
	modified:   l10cloud-formation/lesson10security.yml

iijima-pixcel committed:
modified:   l10cloud-formation/lesson10ap.yml
	modified:   l10cloud-formation/lesson10network.yml
	modified:   l10cloud-formation/lesson10security.yml

iijima-pixcel committed:
modified:   l10cloud-formation/lesson10network.yml
	modified:   l10cloud-formation/lesson10security.yml

iijima-pixcel committed:
modified:   l10cloud-formation/lesson10network.yml
	modified:   l10cloud-formation/lesson10security.yml

iijima-pixcel committed:
modified:   l10cloud-formation/lesson10network.yml
	modified:   l10cloud-formation/lesson10security.yml

MasatoshiMizumoto commented on l10cloud-formation/lesson10security.yml:
```diff
@@ -0,0 +1,74 @@
+      GroupDescription: Security group for SSH HTTP access
+      VpcId: !ImportValue lesson10vpcoutName
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 22
+          ToPort: 22
+          CidrIp: 121.105.112.83/32
```
インバウンドのソースIPは恐らくご自身のインターネットパブリックIPかなと理解しましたので、ここは固定値にしないほうが良いかなと思います！
```suggestion
        - IpProtocol: tcp
          FromPort: 22
          ToPort: 22
          CidrIp: !Ref CiderIpFromInernet
```

iijima-pixcel replied:
Parametersの項目を作り、参照させるようにしました

MasatoshiMizumoto commented on l10cloud-formation/lesson10network.yml:
```diff
@@ -0,0 +1,136 @@
+    Value: !Ref lesson10vpc
+    Export: 
+      Name: lesson10vpcoutName    
+      
```
最終行に改行がない警告が出ているのでファイルを調査＆修正ください！

iijima-pixcel replied:
警告は赤いマイナスマークのことで合っていたでしょうか？
vscodeで作っていたため、改行コードをLFに変更し、ファイル末尾に自動的に改行を挿入する設定にしました。

MasatoshiMizumoto commented on l10cloud-formation/lesson10security.yml:
```diff
@@ -0,0 +1,74 @@
+      Value: !Ref l10rdsDBSecurityGroup
+      Export: 
+        Name: l10rdssecurityoutName    
+        
```
こちらも最終行に改行がない警告が出ているのでファイルを調査＆修正ください！

MasatoshiMizumoto commented on l10cloud-formation/lesson10ap.yml:
```diff
@@ -0,0 +1,117 @@
+        - AssociatePublicIpAddress: "true"
+          DeviceIndex: "0"
+          GroupSet: 
+            - !ImportValue l10ec2securityoutName
```
命名規則がケバブケース、ローワーキャメルケース、パスカルケースどれなのか読み取れなかったので、れにするかは決めた方が良いかなと思います！
例えばローワーキャメルケースだとこう表現します！
```suggestion
            - !ImportValue l10Ec2SecurityOutName
```

MasatoshiMizumoto replied:
私が指摘した部分以外も関係する話ですので、命名規則を決定したら規則に則って全て修正ください！

iijima-pixcel replied:
命名規則をローワーキャメルケースに統一しました。ただし、s3は大文字を使えないため、小文字のままです。

MasatoshiMizumoto commented on l10cloud-formation/lesson10network.yml:
```diff
@@ -0,0 +1,136 @@
+      RouteTableId: !Ref lesson10routeprivate1c
+#Output Parameters
+Outputs:
+  public1aout:
+    Description: subnet from public1al10
+    Value: !Ref public1al10 
+    Export: 
+      Name: public1aoutName   
```
このOutputの書き方だと、中にID(subnet-12345678abcd)が入っているのか、名称（public1al10）が入っているのかExportNameだけみて判断しようとすると誤解するかなと思ったのですが、格納されているのは名称でよかったですか？

iijima-pixcel replied:
中には論理idが入っているので、ExportNameにIdを入れて変更しました

MasatoshiMizumoto reviewed (changes requested):
しっかり設計実装できていると思います！
いくつかコメント入れましたのでご対応お願いします！！

iijima-pixcel committed:
modified:   l10cloud-formation/lesson10ap.yml
	modified:   l10cloud-formation/lesson10network.yml
	modified:   l10cloud-formation/lesson10security.yml
	modified:   lesson10.md

iijima-pixcel committed:
modified:   l10cloud-formation/lesson10ap.yml
	modified:   l10cloud-formation/lesson10network.yml
	modified:   l10cloud-formation/lesson10security.yml

iijima-pixcel committed:
modified:   l10cloud-formation/lesson10ap.yml
	modified:   l10cloud-formation/lesson10network.yml
	modified:   l10cloud-formation/lesson10security.yml

iijima-pixcel committed:
modified:   l10cloud-formation/lesson10ap.yml
	modified:   l10cloud-formation/lesson10network.yml
	modified:   l10cloud-formation/lesson10security.yml

iijima-pixcel committed:
modified:   l10cloud-formation/lesson10ap.yml
	modified:   l10cloud-formation/lesson10network.yml
	modified:   l10cloud-formation/lesson10security.yml

iijima-pixcel committed:
new file:   img/lesson10/ALB-Security.png
	new file:   img/lesson10/Ec2-Security.png
	new file:   img/lesson10/L10-Alb-Tg.png
	new file:   img/lesson10/L10-Alb.png
	new file:   img/lesson10/L10-App-Stack-Resource.png
	new file:   img/lesson10/L10-Ec2.png
	new file:   img/lesson10/L10-Network-Stack-Resource.png
	new file:   img/lesson10/L10-Rds.png
	new file:   img/lesson10/L10-S3.png
	new file:   img/lesson10/L10-Security-Stack-Resource.png
	new file:   img/lesson10/L10-Vpc.png
	new file:   img/lesson10/RDS-Security.png
	deleted:    img/lesson10/l10alb.png
	deleted:    img/lesson10/l10albsecurity.png
	deleted:    img/lesson10/l10albtarget.png
	deleted:    img/lesson10/l10ap-stack-source.png
	deleted:    img/lesson10/l10ec2.png
	deleted:    img/lesson10/l10ec2security.png
	deleted:    img/lesson10/l10net-stack-source.png
	deleted:    img/lesson10/l10rds.png
	deleted:    img/lesson10/l10rdssecurity.png
	deleted:    img/lesson10/l10s3.png
	deleted:    img/lesson10/l10security-stack-source.png
	deleted:    img/lesson10/l10vpc.png
	modified:   lesson10.md

MasatoshiMizumoto reviewed (approved):
LGTM!
